core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -74,7 +74,7 @@
    * Maximum supported data page size (in bytes). In principle, the maximum addressable page size is
    * (1L &lt;&lt; OFFSET_BITS) bytes, which is 2+ petabytes. However, the on-heap allocator's
    * maximum page size is limited by the maximum amount of data that can be stored in a long[]
-   * array, which is (2^32 - 1) * 8 bytes (or 16 gigabytes). Therefore, we cap this at 16 gigabytes.
+   * array, which is (2^31 - 1) * 8 bytes (or 16 gigabytes). Therefore, we cap this at 16 gigabytes.
```

Comment:
"or about 17 gigabytes" is correct, and it's not capped at 16 gigabytes, but the value in the previous sentence. I'd just say "cap at this value".

Reply:
Why be imprecise, if you're bothering to correct other things here? it's closer to 17. It's actually greater than 17.